File: pmb/ai_manage/models/model_config.py
"""模型配置数据模型"""
from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class ModelConfig:
    id: str
    name: str
    provider: str
    api_key: str
    base_url: str
    model_name: str
    is_active: bool = False
    created_at: str = field(default_factory=lambda: datetime.now().isoformat())
    updated_at: str = field(default_factory=lambda: datetime.now().isoformat())

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "name": self.name,
            "provider": self.provider,
            "api_key": self.api_key,
            "base_url": self.base_url,
            "model_name": self.model_name,
            "is_active": self.is_active,
            "created_at": self.created_at,
            "updated_at": self.updated_at,
        }

    @classmethod
    def from_dict(cls, d: dict) -> "ModelConfig":
        return cls(
            id=d.get("id", ""),
            name=d.get("name", ""),
            provider=d.get("provider", ""),
            api_key=d.get("api_key", ""),
            base_url=d.get("base_url", ""),
            model_name=d.get("model_name", ""),
            is_active=d.get("is_active", False),
            created_at=d.get("created_at", ""),
            updated_at=d.get("updated_at", ""),
        )

File: aipmb_manage/lib/models/model_config.dart
class ModelConfig {
  final String id;
  final String name;
  final String provider;
  final String apiKey;
  final String baseUrl;
  final String modelName;
  final bool isActive;
  final String createdAt;
  final String updatedAt;

  ModelConfig({
    required this.id,
    required this.name,
    required this.provider,
    required this.apiKey,
    required this.baseUrl,
    required this.modelName,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ModelConfig.fromJson(Map<String, dynamic> json) {
    return ModelConfig(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      provider: json['provider'] ?? '',
      apiKey: json['api_key'] ?? '',
      baseUrl: json['base_url'] ?? '',
      modelName: json['model_name'] ?? '',
      isActive: json['is_active'] ?? false,
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    'provider': provider,
    'api_key': apiKey,
    'base_url': baseUrl,
    'model_name': modelName,
  };
}